### ISSUE TITLE
Failed to start RxDb (pouchdb-utils): Uncaught ReferenceError: global is not defined

### ISSUE THREAD

**Issue by jega-ms** [pouchdb/pouchdb#7984]
Failed to start the rxDb on reactJs-typescript project. getting the error "global is not defined" 

**Environment**

- ReactJs , Typescript , RXDB

- Electron 

 

```
Uncaught ReferenceError: global is not defined
    at Object../node_modules/immediate/lib/browser.js (browser.js:2)
    at __webpack_require__ (bootstrap:726)
    at fn (bootstrap:100)
    at Module../node_modules/pouchdb-utils/lib/index-browser.es.js (index-browser.es.js:1)
    at __webpack_require__ (bootstrap:726)
    at fn (bootstrap:100)
    at Module../node_modules/pouchdb-selector-core/lib/index.es.js (index.es.js:1)
    at __webpack_require__ (bootstrap:726)
    at fn (bootstrap:100)
    at Module../node_modules/rxdb/dist/es/query-change-detector.js (query-change-detector.js:1)

```

```
import * as RxDB from 'rxdb';
import { RxDatabase, RxJsonSchema } from 'rxdb';

const contactsSchema: RxJsonSchema<any> = {
    title: 'human schema',
    description: 'describes a human being',
    version: 0,
    keyCompression: true,
    type: 'object',
    properties: {
        passportId: {
            type: 'string',
            primary: true
        },
        firstName: {
            type: 'string'
        },
        lastName: {
            type: 'string'
        },
        age: {
            type: 'integer'
        }
    },
    required: ['firstName', 'lastName']
};

export class DatabaseManager {
    private db: RxDatabase | undefined;
    constructor() {
        console.log('DatabaseManager loaded....');
    }
    private async createDB() {
        this.db = await RxDB.create({
            name: 'messages',
            multiInstance: false,
            adapter: "websql"
        })
    }
    private async createCollection() {
        if (this.db) {
            await this.db.collection({
                name: 'contacts',
                schema: contactsSchema
            })
        }
    }
    private async tryDump() {
        if (this.db) {
            let d = await this.db.dump();
            console.log(`dump = ${JSON.stringify(d)}`);
        }
    }
    private async tryInsert() {
        if (this.db) {
            this.db['contacts'].insert({ firstname: 'Jo', lastNama: 'blue', mobile: '+192532400000' });
        }
    }

    async testRxDB() {
        console.log('... testRxDB');
        await this.createDB();
        await this.createCollection();
        await this.tryInsert();
        await this.tryDump();
    }
}

```

**Comment by pubkey** [comment]
PouchDB and so also RxDB rely on the `global` variable of nodejs. This is polyfilled by some frameworks but is no longer since angular 5.
You have to add the polyfill by yourself like [here](https://github.com/pubkey/rxdb/blob/master/examples/angular/src/polyfills.ts)

**Comment by MiguelQueiroz** [comment]
I could solve it with index.html

  <script>
    window.process = window.process || {env: {NODE_DEBUG:undefined, DEBUG:undefined}};
    window.global = window;
  </script>

and:
npm install pouchdb-browser
npm install events

**Comment by lokeshparmar998** [comment]
I am trying to use pouchdb with nuxt-3 and faced the same error **"global is not defined"**. I fixed it by changing the builder in nuxt.config.ts from vite (default builder) to webpack 5.

`export default defineNuxtConfig({
    builder: 'webpack'
})`

Do not forget to install `@nuxt/webpack-builder` as a dev dependency.

And import it like this for typescript: `import * as PouchDB from '@/node_modules/pouchdb/dist/pouchdb'`

**Comment by minitauros** [comment]
I did what @MiguelQueiroz proposed.

> I could solve it with index.html
> 
> <script> window.process = window.process || {env: {NODE_DEBUG:undefined, DEBUG:undefined}}; window.global = window; </script>

Then I followed some advice of [this Stackoverflow post](https://stackoverflow.com/questions/60779965/typeerror-superctor-is-undefined-inherits-inherits-browser-js):

```
yarn add stream-browserify
```

and then add the following to vite.config.ts

```
  resolve: {
    alias: {
      'stream': 'stream-browserify',
    },
  },
```

Seems like one of the RxDB modules (probably pouchdb) relies on `stream`, which targets Node, not the browser. `stream-browserify`, as they mention [on their GitHub](https://github.com/browserify/stream-browserify), is "the stream module from node core, for browsers!".
